File: digits-multiplication.py
import numpy


def checkio(number):
    return numpy.prod([float(i) for i in str(number) if 0 < int(i)], dtype=numpy.int32).item(0)


# These "asserts" using only for self-checking and not necessary for auto-testing
if __name__ == '__main__':
    assert checkio(123405) == 120
    assert checkio(999) == 729
    assert checkio(1000) == 1
    assert checkio(1111) == 1
    print("Coding complete? Click 'Check' to review your tests and earn cool rewards!")

File: js/Elementary/digits_multiplication.js
#!/usr/bin/env checkio --domain=js run digits-multiplication

// https://js.checkio.org/mission/digits-multiplication/

// You are given a positive integer.    Your function should calculate the product of the digits excluding any zeroes.
// 
// For example: The number given is 123405. The result will be 1*2*3*4*5=120 (don't forget to exclude zeroes).
// 
// Input:A positive integer.
// 
// Output:The product of the digits as an integer.
// 
// Precondition:0 < number < 106
// 
// 
// END_DESC

"use strict";

function digitsMultip(data) {
    return 0;
}

var assert = require('assert');

if (!global.is_checking) {
    console.log('Example:')
    console.log(digitsMultip(123405))
    
    assert.equal(digitsMultip(123405), 120, "1st");
    assert.equal(digitsMultip(999), 729, "2nd");
    assert.equal(digitsMultip(1000), 1, "3rd");
    assert.equal(digitsMultip(1111), 1, "4th");
    console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");
}